src/CodeStyle/Core/CodeFixes/DocumentBasedFixAllProvider.cs

```diff
@@ -75,41 +68,28 @@ private async Task<Document> GetDocumentFixesAsync(FixAllContext fixAllContext)
                 return fixAllContext.Document;
             }
 
-            var newRoot = await FixAllInDocumentAsync(fixAllContext, fixAllContext.Document, diagnostics).ConfigureAwait(false);
-            if (newRoot == null)
-            {
-                return fixAllContext.Document;
-            }
-
-            return fixAllContext.Document.WithSyntaxRoot(newRoot);
+            return await FixAllInDocumentAsync(fixAllContext, fixAllContext.Document, diagnostics).ConfigureAwait(false);
         }
 
         private async Task<Solution> GetSolutionFixesAsync(FixAllContext fixAllContext, ImmutableArray<Document> documents)
         {
             var documentDiagnosticsToFix = await FixAllContextHelper.GetDocumentDiagnosticsToFixAsync(fixAllContext, progressTrackerOpt: null, (document, cancellationToken) => false).ConfigureAwait(false);
 
             var solution = fixAllContext.Solution;
-            var newDocuments = new List<Task<SyntaxNode>>(documents.Length);
+            var newDocuments = new Dictionary<DocumentId, Task<Document>>(documents.Length);
             foreach (var document in documents)
             {
-                if (!documentDiagnosticsToFix.TryGetValue(document, out var diagnostics))
+                if (documentDiagnosticsToFix.TryGetValue(document, out var diagnostics))
                 {
-                    newDocuments.Add(document.GetSyntaxRootAsync(fixAllContext.CancellationToken));
-                    continue;
+                    newDocuments.Add(document.Id, FixAllInDocumentAsync(fixAllContext, document, diagnostics));
                 }
-
-                newDocuments.Add(FixAllInDocumentAsync(fixAllContext, document, diagnostics));
             }
 
-            for (var i = 0; i < documents.Length; i++)
+            foreach (var documentIdAndTask in newDocuments)
             {
-                var newDocumentRoot = await newDocuments[i].ConfigureAwait(false);
-                if (newDocumentRoot == null)
-                {
-                    continue;
-                }
-
-                solution = solution.WithDocumentSyntaxRoot(documents[i].Id, newDocumentRoot);
+                var newDocument = await documentIdAndTask.Value.ConfigureAwait(false);
```

Comment:
based on convos with some other devs, i think you first want to do `await Task.WhenAll(newDocuments.Values)...`.  That way if any particular task throws an exception you don't lose it.